src/Entities/Player.cpp

```diff
@@ -1998,8 +2000,7 @@ bool cPlayer::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 			a_OldWorld.GetName().c_str(), a_World->GetName().c_str(),
 			ParentChunk->GetPosX(), ParentChunk->GetPosZ()
 		);
-		ParentChunk->RemoveEntity(this);
-		a_World->AddPlayer(this, &a_OldWorld);  // New world will take over and announce client at its next tick
+		a_World->AddPlayer(std::unique_ptr<cPlayer>(static_cast<cPlayer *>(ParentChunk->RemoveEntity(*this).release())), &a_OldWorld);  // New world will take over and announce client at its next tick
```

Comment:
Because someone broke inheritance by making players specially need AddPlayer rather than AddEntity, and I couldn't get downcasting working with unique pointers.